src/System.IO.Ports/tests/SerialPort/ReadExisting_Generic.cs

```diff
@@ -170,13 +168,7 @@ public void ParityErrorOnLastByte()
 
                 com2.Write(bytesToWrite, 0, bytesToWrite.Length);
 
-                waitTime = 0;
-
-                while (bytesToWrite.Length + 2 > com1.BytesToRead && waitTime < 500)
-                {
-                    System.Threading.Thread.Sleep(50);
-                    waitTime += 50;
-                }
+                TCSupport.WaitForReadBufferToLoad(com1, bytesToWrite.Length);
```

Comment:
Well spotted, but this is correcting what I assume was a cut-and-paste bug in the original test (there aren't bytesToWrite+2 bytes actually written in this test).  Originally a timeout from this loop was ignored, so it didn't actually matter that the loop termination condition was wrong.